### ISSUE TITLE
Integration "@astrojs/solid-js" does not provide "vite" to "vite-plugin-solid"

### ISSUE THREAD

**Issue by marbrex** [withastro/astro#12460]
### Astro Info

```block
Astro                    v4.16.13
Node                     v22.3.0
System                   Windows (x64)
Package Manager          yarn
Output                   static
Adapter                  none
Integrations             @astrojs/solid-js
```

### Describe the Bug

I have a peer dependency problem. An Astro integration package **`@astrojs/solid-js`** does not provide **`vite`** to **`vite-plugin-solid`**. Astro fails to run with the message "_Unable to load your Astro config_". That is because the above mentionned SolidJS integration is used in the Astro config (`astro.config.js`). If I add **`vite`** to the **dev** dependencies of my project, the peer dependency gets resolved and I no longer have the issue, Astro works just fine. However, is that the right way of resolving the peer dependency ? Or would it be more appropriate to include **`vite`** in `dependencies` of **`@astrojs/solid-js`** package instead of `devDependencies` as it is now ?  
 https://github.com/withastro/astro/blob/25baa4ed0c5f55fa85c2c7e2c15848937ed1dc9b/packages/integrations/solid/package.json#L37-L45  
 
> Btw, I am using Yarn Berry `yarn@4.5.1`. I noticed that `npm` does not produce the error, so it seems like it's related only for `yarn`. May be there is something in the way how `npm` deals with the transitive dependencies.  

### What's the expected result?

The peer dependency is correctly resolved.  

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-ygobhj?file=package.json

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by ematipico** [comment]
I'm not really sure what's the issue, without a valid reproduction (it works?). But the integration says that `solid-js` is needed a peer dependency, so the project must install it 

https://github.com/withastro/astro/blob/25baa4ed0c5f55fa85c2c7e2c15848937ed1dc9b/packages/integrations/solid/package.json#L46-L49

**Comment by marbrex** [comment]
@ematipico , thanks for your reply. In fact, the problem comes from the **`vite-plugin-solid`**'s peer dependencies, which is used in the integration. The **`vite`** package, to be precise. Tthe integration does not provide **`vite`** to **`vite-plugin-solid`**. It seems like the error occurs when using **Yarn in PnP mode**.  

StackBlitz doesn't support Corepack, and hence Yarn PnP. So I've created a repo on GitHub ([marbrex/astro-solid-yarn](https://github.com/marbrex/astro-solid-yarn)) with a valid reproduction of the problem.  

Steps to reproduce the issue :
1) Clone the repo [marbrex/astro-solid-yarn](https://github.com/marbrex/astro-solid-yarn)  
2) Run `yarn run dev`  

> P.S. If you want to view the project :  
> 1) Open the project in Visual Studio Code  
> 2) Use the project's TypeScript (VS Code should suggest you automatically).

**Comment by marbrex** [comment]
To work around this issue, I added `vite` to my project's `devDependencies`, which resolved the peer dependency problem. However, I noticed that `astro` already has `vite` in its dependencies, so it seems redundant to explicitly add `vite` in my project.  

Additionally, I tested a Yarn-specific workaround using the `packageExtensions` field in `.yarnrc.yml` to extend `@astrojs/solid-js` dependencies like this :  

```yml
packageExtensions:
  '@astrojs/solid-js@^4.4.3':
    dependencies:
      vite: '*'
```

While this resolves the issue temporarily, the real fix should be made in the `@astrojs/solid-js` package itself by correctly defining `vite` as a dependency in its `package.json`.
